sdk/core/core-rest-pipeline/test/defaultRetryPolicy.spec.ts

```diff
@@ -36,26 +36,29 @@ describe("defaultRetryPolicy", function () {
   });
 
   it("It should give up after the default maxRetries is reached", async () => {
+    const errorLists = ["ETIMEDOUT", "ESOCKETTIMEDOUT", "ECONNREFUSED", "ENOENT", "ENOTFOUND"];
     const request = createPipelineRequest({
       url: "https://bing.com",
     });
-    const testError = new RestError("Test Error!", { code: "ENOENT" });
-
     const policy = defaultRetryPolicy();
-    const next = sinon.stub<Parameters<SendRequest>, ReturnType<SendRequest>>();
-    next.rejects(testError);
-
     const clock = sinon.useFakeTimers();
 
-    let catchCalled = false;
-    const promise = policy.sendRequest(request, next);
-    promise.catch((e) => {
-      catchCalled = true;
-      assert.strictEqual(e, testError);
-    });
-    await clock.runAllAsync();
-    assert.strictEqual(next.callCount, DEFAULT_RETRY_POLICY_COUNT + 1);
-    assert.isTrue(catchCalled);
+    for (const errorCode of errorLists) {    
+      let testError = new RestError("Test Error!", { code: errorCode });
+
+      let next = sinon.stub<Parameters<SendRequest>, ReturnType<SendRequest>>();
+      next.rejects(testError);
+
+      let catchCalled = false;
+      let promise = policy.sendRequest(request, next);
```

Comment:
Should these lets other than `catchCalled` be consts?

Reply:
Yes, that's the failure in the pipeline too. Fixing in the new commit